app/assets/src/components/SampleUpload.jsx

```diff
@@ -391,13 +392,24 @@ class SampleUpload extends React.Component {
     }
   }
 
-  baseName(str) {
-    let base = new String(str).substring(str.lastIndexOf("/") + 1);
-    if (base.lastIndexOf(".") != -1) {
+  baseName = str => {
+    let base = cleanLocalFileName(str);
+
+    let separator = "/";
+    if (base.includes("\\")) {
+      // If the name includes the backslash \ it's probably from Windows.
+      separator = "\\";
+    }
+
+    // Get the last piece
+    base = base.substring(base.lastIndexOf(separator) + 1);
```

Comment:
I think you can do `last(base.split(separator))` with `import last from "lodash/fp"`